shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCXShuffleTransport.scala

```diff
@@ -240,21 +241,27 @@ class UCXShuffleTransport(shuffleServerId: BlockManagerId, rapidsConf: RapidsCon
   }
 
   // NOTE: this pool, as is, will add a new thread per task. This will likely change.
-  private[this] val clientExecutor = new ThreadPoolExecutor(1,
-    rapidsConf.shuffleMaxClientThreads,
-    rapidsConf.shuffleClientThreadKeepAliveTime,
-    TimeUnit.SECONDS,
-    new ArrayBlockingQueue[Runnable](1),
-    GpuDeviceManager.wrapThreadFactory(
-      new ThreadFactoryBuilder()
-        .setNameFormat("shuffle-transport-client-exec-%d")
-        .setDaemon(true)
-        .build,
-      null,
-      () => RmmSpark.removeAllCurrentThreadAssociation()),
+  private[this] val clientExecutor = {
+    // We are using the TrampolineUtil to create a ThreadPoolExecutor even though we are
+    // resetting most of the params that were set using the ThreadUtils class.
+    // The reason is that this way we will get the Hadoop conf necessary for Unity Catalog
+    val threadPoolExecutor = TrampolineUtil.newDaemonCachedThreadPool("ucx-shuffle-transport",
+      rapidsConf.shuffleMaxClientThreads,
+      rapidsConf.shuffleClientThreadKeepAliveTime)
+    threadPoolExecutor.setCorePoolSize(1)
+    threadPoolExecutor.setThreadFactory(
+      GpuDeviceManager.wrapThreadFactory(
+        new ThreadFactoryBuilder()
+          .setNameFormat("shuffle-transport-client-exec-%d")
+          .setDaemon(true)
+          .build,
+        null,
+        () => RmmSpark.removeAllCurrentThreadAssociation()))
```

Comment:
Ignoring the existing thread factory will likely defeat the point of using ThreadUtils to create the thread pool.  Any per-thread logic provided by the Spark platform is highly likely placed in the thread factory that ThreadUtils is setting up.
```suggestion
    threadPoolExecutor.setThreadFactory(
      GpuDeviceManager.wrapThreadFactory(
        threadPoolExecutor.getThreadFactory,
        null,
        () => RmmSpark.removeAllCurrentThreadAssociation()))
```